plugins/lime-plugin-pirania/pirania.spec.js

```diff
@@ -0,0 +1,145 @@
+import { h } from "preact";
+import { fireEvent, screen } from "@testing-library/preact";
+import { render } from "utils/test_utils";
+import "@testing-library/jest-dom";
+import waitForExpect from "wait-for-expect";
+import { route } from "preact-router";
+import VoucherList from "./src/screens/voucherList";
+import { listVouchers } from "./src/piraniaApi";
+
+jest.mock("./src/piraniaApi");
+const now = new Date().getTime() / 1000;
+let vouchers = [
+	{
+		code: "NNDAMD",
+		id: "fteNhN",
+		is_active: false,
+		name: "hiure2",
+		duration_m: 10000000000,
+		creation_date: 1631880815,
+		permanent: false,
+		status: "available",
+		author_node: "conteiner"
+	},
+	{
+		mac: "6c:88:14:ba:c4:84",
+		expiration_date: 601631880811,
+		code: "BAVWNS",
+		id: "5nLNT9",
+		activation_date: 1631880811,
+		name: "hiure2",
+		duration_m: 10000000000,
+		is_active: true,
+		creation_date: 1631880790,
+		permanent: false,
+		status: "used",
+		author_node: "conteiner"
+	},
+	{
+		mac: "98:9e:63:a7:91:e2",
+		expiration_date: 6e25,
+		code: "COSVQK",
+		id: "6TIgTr",
+		activation_date: 1633530122,
+		name: "luandro-iphone",
+		duration_m: 1e24,
+		is_active: true,
+		creation_date: 1633530042,
+		permanent: false,
+		status: "disabled",
+		author_node: "sede"
+	},
+	{
+		code: "PIDFIG",
+		id: "x5crd4",
+		activation_date: now - 600e3,
+		name: "luandro",
+		duration_m: 1200,
+		is_active: false,
+		creation_date: now - 700e3,
+		permanent: false,
+		status: "available",
+		author_node: "pirania"
+	}
+];
+
+describe("voucher list", () => {
+	it("shows a button to create a new voucher that routes to creation page", async () => {
+		render(<VoucherList vouchers={vouchers} />);
```

Comment:
Let please keep the pattern of testing components that render entire screen, having no properties passed by beside url parameters, and mocking backend apis.
That let us keep track of how things connect to each other.

In this case, we could test PiraniaPage instead of VoucherList and mock getVouchersList api for each case.